Sources/SpeziViews/Views/Layout/HorizontalGeometryReader.swift

```diff
@@ -39,7 +39,13 @@ public struct HorizontalGeometryReader<Content: View>: View {
                 }
             )
             .onPreferenceChange(WidthPreferenceKey.self) { width in
-                self.width = width
+                // The `onPreferenceChange` view modfier now takes a `@Sendable` closure, therefore we cannot capture `@MainActor` isolated properties
+                // on the `View` directly anymore: https://developer.apple.com/documentation/swiftui/view/onpreferencechange(_:perform:)?changes=latest_minor
+                // However, as the `@Sendable` closure is still run on the MainActor (at least in my testing on 18.2 RC SDKs), we can use `MainActor.assumeIsolated`
+                // to avoid scheduling a `MainActor` `Task`, which could delay execution and cause unexpected UI behavior.
+                MainActor.assumeIsolated {
```

Comment:
if you really wanna be safe you might instead want to do

```swift
if Thread.isMainThread {
    MainActor.assumeIsolated {
        self.width = width
    }
} else {
    Task { @MainActor in
        self.width = width
    }
}
```

that way at least it won't crash should they decide to move it off the main actor in a future release, without documenting that change

Reply:
I like that solution @lukaskollmer, gives us way more safety, let's go for that one!